### PR TITLE
Internal go-yaml/yaml fork

### PR DESCRIPTION
**Pull request by KnVerey**
@natasha41575 here's a script we can use to make an internal copy of yaml.v3 with your changes applied. It pulls down the latest version from upstream and then applies your commit instead of pulling your branch directly. That's not strictly necessary now, since your changes are based on the same commit k/k is using. But if we need to merge this, we could end up in a situation where your changes need to be rebased to a version that is ahead of k/k (as was the case when Jeff was working on his version of the changes).

To use this: 
- update the script's `CHERRY_PICK_SHA` if necessary (e.g. if the go-yaml PR changes)
- run the script (it makes commits for you)
- cherry-pick the fix commits from #4004 

ALLOW_MODULE_SPAN

### CODE REVIEW

File: kyaml/internal/forked/README.md [hunk 1]
```diff
@@ -0,0 +1,11 @@
+# kyaml internal forks
+
+## qri-io/starlib
+
+This code is used by the starlark runtime. We copied it in to reduce the dependencies being brought over to kubectl by the kustomize integration. Should it need updating, do so via manual copy-paste.
+
+## go-yaml/yaml
+
+This code is used extensively by kyaml. It is a copy of upstream at a particular revision that kubectl is using, with [a change we need](https://github.com/go-yaml/yaml/pull/746) cherry-picked on top. For context on why the cherry-pick to an old revision is necessary, see https://github.com/kubernetes-sigs/kustomize/issues/3946.
```
Comment (KnVerey) [inline kyaml/internal/forked/README.md:9]: Note to self: update this last sentence if we use this--it's not to an old revision anymore

File: kyaml/internal/forked/update-go-yaml.sh [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+    printf "\n\n%s\n" "${blue}$1${normal}"
+}
+
+# This should be the version of go-yaml v3 used by kubectl
+# In the original fork, this is 496545a6307b2a7d7a710fd516e5e16e8ab62dbc
+export GOYAML_SHA=$1
+export GOYAML_REF="goyaml-$GOYAML_SHA"
+
+# This is the sha of the commit in https://github.com/go-yaml/yaml/pull/753
+export CHERRY_PICK_SHA=e965076f45946b0bda907e6c2316b4523207630e
```
Comment (natasha41575) [inline kyaml/internal/forked/update-go-yaml.sh:28]: maybe a dumb question, but where did this sha come from?

Comment (KnVerey) [reply kyaml/internal/forked/update-go-yaml.sh:28]: Your PR 😄 
This needs to be updated once your PR is finalized. Right now this is the revision before you made some naming changes yesterday.

File: kyaml/yaml/alias.go [hunk 3]
```diff
@@ -33,7 +43,10 @@ var Unmarshal = yaml.Unmarshal
 var NewDecoder = yaml.NewDecoder
 var NewEncoder = func(w io.Writer) *yaml.Encoder {
 	e := yaml.NewEncoder(w)
-	e.SetIndent(2)
+	e.SetIndent(indent)
+	if sequenceIndentationStyle == CompactSequenceStyle {
```
Comment (phanimarupaka) [inline kyaml/yaml/alias.go:47]: What is the reason behind not making this configurable ? Will it be exposed in the future ?

Comment (phanimarupaka) [reply kyaml/yaml/alias.go:47]: kpt doesn't hold strong opinion about seq indentation, but we want minimum disruption to the users. The plan is to identify the existing indentation of package resources and retain it, for which, we need this to be configurable. Let me know your thoughts.

As of now, kpt can't upgrade to latest kyaml v0.11.0

Comment (natasha41575) [reply kyaml/yaml/alias.go:47]: I filed https://github.com/kubernetes-sigs/kustomize/issues/4029. @KnVerey PTAL when you can

**Pull request merged**